persist/sqlite/accounts_test.go

```diff
@@ -0,0 +1,165 @@
+package sqlite
+
+import (
+	"path/filepath"
+	"testing"
+	"time"
+
+	proto4 "go.sia.tech/core/rhp/v4"
+	"go.sia.tech/core/types"
+	rhp4 "go.sia.tech/coreutils/rhp/v4"
+	"go.sia.tech/hostd/host/contracts"
+	"go.sia.tech/hostd/index"
+	"go.uber.org/zap/zaptest"
+	"lukechampine.com/frand"
+)
+
+func TestRHP4Accounts(t *testing.T) {
+	log := zaptest.NewLogger(t)
+	db, err := OpenDatabase(filepath.Join(t.TempDir(), "test.db"), log)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer db.Close()
+
+	renterKey := types.NewPrivateKeyFromSeed(frand.Bytes(32))
+	hostKey := types.NewPrivateKeyFromSeed(frand.Bytes(32))
+
+	c, count, err := db.V2Contracts(contracts.V2ContractFilter{})
+	if err != nil {
+		t.Fatal(err)
+	} else if len(c) != 0 {
+		t.Fatal("expected no contracts")
+	} else if count != 0 {
+		t.Fatal("expected no contracts")
+	}
+
+	// add a contract to the database
+	contract := contracts.V2Contract{
+		ID: frand.Entropy256(),
+		V2FileContract: types.V2FileContract{
+			RenterPublicKey:  renterKey.PublicKey(),
+			HostPublicKey:    hostKey.PublicKey(),
+			ProofHeight:      100,
+			ExpirationHeight: 200,
+		},
+	}
+
+	if err := db.AddV2Contract(contract, rhp4.TransactionSet{}); err != nil {
+		t.Fatal(err)
+	}
+
+	checkMetricConsistency := func(t *testing.T, potential, earned proto4.Usage) {
+		m, err := db.Metrics(time.Now())
+		if err != nil {
+			t.Fatal(err)
+		}
+		switch {
+		case m.Revenue.Potential.Ingress != potential.Ingress:
+			t.Fatalf("expected potential ingress %v, got %v", potential.Ingress, m.Revenue.Potential.Ingress)
+		case m.Revenue.Potential.Egress != potential.Egress:
+			t.Fatalf("expected potential egress %v, got %v", potential.Egress, m.Revenue.Potential.Egress)
+		case m.Revenue.Potential.Storage != potential.Storage:
+			t.Fatalf("expected storage %v, got %v", potential.Storage, m.Revenue.Potential.Storage)
+		case m.Revenue.Potential.RPC != potential.RPC:
+			t.Fatalf("expect RPC %v, got %v", potential.RPC, m.Revenue.Potential.RPC)
+		case m.Revenue.Earned.Ingress != earned.Ingress:
+			t.Fatalf("expected ingress %v, got %v", earned.Ingress, m.Revenue.Earned.Ingress)
+		case m.Revenue.Earned.Egress != earned.Egress:
+			t.Fatalf("expected egree %v, got %v", earned.Egress, m.Revenue.Earned.Egress)
+		case m.Revenue.Earned.Storage != earned.Storage:
+			t.Fatalf("expected storage %v, got %v", earned.Storage, m.Revenue.Earned.Storage)
+		case m.Revenue.Earned.RPC != earned.RPC:
+			t.Fatalf("expected RPC %v, got %v", earned.RPC, m.Revenue.Earned.RPC)
+		}
+	}
+
+	sk := types.GeneratePrivateKey()
+	account := proto4.Account(sk.PublicKey())
+
+	// deposit funds
```

Comment:
I think `RHP4AccountBalance` is not called at all and we should probably also test the edge cases of withdrawing from an account and fetching the balance of an account before it even exists.